tbot_contrib/gpio/gpio.py

```diff
@@ -0,0 +1,60 @@
+from tbot.machine import linux
+
+
+class gpio:
+    def __init__(self, lnx: linux.LinuxShell, gpio_number: str):
+        self._lnx = lnx
+        self._gpio_number = gpio_number
+        self._gpio_sysclass_path = "/sys/class/gpio/"
+        self._gpio = self._gpio_sysclass_path + "gpio" + self._gpio_number
+        self._export()
+
+    def _export(self) -> None:
+        output = self._lnx.exec0(
+            "find", self._gpio_sysclass_path, "-name", "gpio" + self._gpio_number
+        )
+        if output.strip() == (self._gpio):
+            pass
+        else:
+            self._lnx.exec0(
+                "echo",
+                self._gpio_number,
+                linux.Raw(">"),
+                self._gpio_sysclass_path + "export",
+            )
+
+    def _is_in_direction(self) -> bool:
+        output = self._lnx.exec0("cat", self._gpio + "/direction")
+        if output.strip() == "in":
+            return True
+        else:
+            return False
+
+    def set_in_direction(self) -> None:
+        if self._is_in_direction():
+            pass
+        else:
+            self._lnx.exec0("echo", "in", linux.Raw(">"), self._gpio + "/direction")
+
+    def set_out_direction(self) -> None:
+        if not self._is_in_direction():
+            pass
+        else:
+            self._lnx.exec0("echo", "out", linux.Raw(">"), self._gpio + "/direction")
+
+    def get_direction(self) -> str:
+        return self._lnx.exec0("cat", self._gpio + "/direction").strip()
+
+    def set_value(self, value: str) -> None:
+        if self._is_in_direction():
+            self.set_out_direction()
```

Comment:
I'm not sure whether this is something we really want.  Automatically switching the direction is a bit dangerous, because it happens hidden from the programmer (and might damage hardware in some cases).  I'd prefer forcing users to explicitly set the direction.